common_tools/instruction_count.py

```diff
@@ -0,0 +1,53 @@
+import os
+import argparse
+import csv
+
+def count_instr_in_file(file_path):
+    counting = False
+    instr_count = 0
+    
+    with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:
+        for line in file:
+            # Start counting when "Kernarg preload header" is read
+            if "Kernarg preload header" in line:
+                counting = True
+            # End when "---" is read
```

Comment:
We don't need this IMO, it should be enough to enumerate lines and return when you see `s_endprgm`

Reply:
but this would include the header, e.g.:
```
	.text
	.amdgcn_target "amdgcn-amd-amdhsa--gfx942"
	.amdhsa_code_object_version 5
	.globl	main_dispatch_0_matmul_transpose_b_2x3584x8192_bf16
	.p2align	8
	.type	main_dispatch_0_matmul_transpose_b_2x3584x8192_bf16,@function
main_dispatch_0_matmul_transpose_b_2x3584x8192_bf16:
```
maybe @function would be more general?

either way works for me. I'll have it count from BOF for now.

Reply:
that's fine